lib/SourceMapDevToolPlugin.js

```diff
@@ -479,6 +479,34 @@ class SourceMapDevToolPlugin {
 												"\n/*$1*/"
 											);
 									}
+
+									if (options.debugIds) {
+										// We need a uuid which is 128 bits so we need 2x 64 bit hashes.
+										// The first 64 bits is a hash of the source.
+										const sourceHash = createHash("xxhash64")
+											.update(source)
+											.digest("hex");
+										// The next 64 bits is a hash of the filename and sourceHash
+										const hash128 = `${sourceHash}${createHash("xxhash64")
```

Comment:
https://webpack.js.org/configuration/output/#outputhashfunction

Reply:
The same for `sourceHash` and respect other options, like https://webpack.js.org/configuration/output/#outputhashsalt

Reply:
Is there a reason to not just use a specific (maybe the fastest) hash here? 

The code relies on getting a specific number of hex characters and if users pass a custom `hashFunction`, will this always be the case?

Reply:
> Is there a reason to not just use a specific (maybe the fastest) hash here?

maybe people want to change the logic depending on the options provided

Reply:
> the `hashFunction` can now be a constructor to a custom hash function. You can provide a non-crypto hash function for performance reasons.

My understanding is that users can supply a custom `hashFunction` if they want to change the way webpack generates file name hashes.

For Debug IDs we almost certainly don't want users to be able to configure the hash function. According to the spec, Debug ids should be a uuid derived from a deterministic hash of the file source code. The hash type isn't defined in the spec but it should be deterministic and it should have enough entropy that it is effectively unique. If we allow users to configure the hash function, the debug IDs may no longer meet the spec.

In the PR code, the 128bit debug ID is created from two `xxhash64` hashes and on my machine it does this with a 300kB bundle in 2 milliseconds.